datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java

```diff
@@ -227,6 +231,12 @@ private String getAncestorTablePath(CarbonTable currentTable) {
         }
       }
     }
+    if (hitBlocklets == null) {
+      LOGGER.warn(String.format("HitBlocklets is empty in bloom filter prune method. " +
```

Comment:
Is this a potential problem? If not, why output 'warn' message?

Reply:
Yes, it will not be executed normally ,when filteredShard's size is processed, but for protection, adding this code

Reply:
can you make it non-null, such as making it an empty set

Reply:
@jackylk empty sets present all blocklets are pruned, and null presents pruning is not effective,so it should return null.
Now out datamap lack null processing（exception scene），it need to be enhanced later